osu.Game.Rulesets.Taiko/Difficulty/Skills/Colour.cs

```diff
@@ -49,95 +29,8 @@ public Colour(Mod[] mods)
 
         protected override double StrainValueOf(DifficultyHitObject current)
         {
-            // changing from/to a drum roll or a swell does not constitute a colour change.
-            // hits spaced more than a second apart are also exempt from colour strain.
-            if (!(current.LastObject is Hit && current.BaseObject is Hit && current.DeltaTime < 1000))
-            {
-                monoHistory.Clear();
-
-                var currentHit = current.BaseObject as Hit;
-                currentMonoLength = currentHit != null ? 1 : 0;
-                previousHitType = currentHit?.Type;
-
-                return 0.0;
-            }
-
-            var taikoCurrent = (TaikoDifficultyHitObject)current;
-
-            double objectStrain = 0.0;
-
-            if (previousHitType != null && taikoCurrent.HitType != previousHitType)
-            {
-                // The colour has changed.
-                objectStrain = 1.0;
-
-                if (monoHistory.Count < 2)
-                {
-                    // There needs to be at least two streaks to determine a strain.
-                    objectStrain = 0.0;
-                }
-                else if ((monoHistory[^1] + currentMonoLength) % 2 == 0)
-                {
-                    // The last streak in the history is guaranteed to be a different type to the current streak.
-                    // If the total number of notes in the two streaks is even, nullify this object's strain.
-                    objectStrain = 0.0;
-                }
-
-                objectStrain *= repetitionPenalties();
-                currentMonoLength = 1;
-            }
-            else
-            {
-                currentMonoLength += 1;
-            }
-
-            previousHitType = taikoCurrent.HitType;
-            return objectStrain;
+            double difficulty = ((TaikoDifficultyHitObject)current).Colour?.EvaluatedDifficulty ?? 0;
```

Comment:
Another important point to consider is that objects after the "first object in a grouping" are also getting significant strain values associated with them by virtue of the very slow strain decay - I would consider looking into ways of not associating strain with unimportant objects at all if that sounds of value to you.

Reply:
We are looking into ways of strain in [this](https://github.com/vunyunt/osu/tree/taiko-strain-decay-skill) branch, but decided to keep it out of this pr due to its already sizeable scope. But good catch, that wasn't something I entirely considered as an aftereffect.

Reply:
Interesting idea, it might be a good reason to try out the alternative to pre-evaluation mentioned earlier. Though this was a desired behaviour - since we are combining skill per-strain (see Peaks.cs), having colour value instantly drop to zero on any mono notes may be undesirable.